nitric/proto/keyvalue/v1/keyvalue.proto

```diff
@@ -0,0 +1,77 @@
+syntax = "proto3";
+package nitric.proto.KeyValue.v1;
+
+import "google/protobuf/struct.proto";
+
+// protoc plugin options for code generation
+// option go_package = "nitric/v1;v1";
+option go_package = "github.com/nitrictech/nitric/core/pkg/proto/keyvalue/v1;KeyValuepb";
+option java_package = "io.nitric.proto.keyvalue.v1";
+option java_multiple_files = true;
+option java_outer_classname = "KeyValue";
+option php_namespace = "Nitric\\Proto\\KeyValue\\V1";
+option csharp_namespace = "Nitric.Proto.KeyValue.v1";
+
+// Service for storage and retrieval of simple JSON keyValue
+service KeyValue {
+  // Get an existing document
+  rpc Get (KeyValueGetRequest) returns (KeyValueGetResponse);
+
+  // Create a new or overwrite an existing document
+  rpc Set (KeyValueSetRequest) returns (KeyValueSetResponse);
+
+  // Delete an existing document
+  rpc Delete (KeyValueDeleteRequest) returns (KeyValueDeleteResponse);
+}
+
+// Provides a Collection type for storing documents
+message Store {
+  // The store name
+  string name = 1;
+}
+
+// Provides a document identifying key type
+message Key {
+  // The item collection
+  string store = 1;
+
+  // The items unique key
+  string key = 2;
+}
+
+// Provides a return document type
+message Value {
+  // The document's unique key, including collection/sub-collections
+  Key key = 1;
+
+  // The document content (JSON object)
+  google.protobuf.Struct content = 2;
+}
+
+// Service Request & Response Messages
+
+message KeyValueGetRequest {
+  // Key of the document to retrieve
+  Key key = 1;
+}
+
+message KeyValueGetResponse {
+  // The retrieved value
+  Value value = 1;
+}
+
+message KeyValueSetRequest {
+  // Key of the document to set
+  Key key = 1 ;
+  // The document content to store (JSON object)
+  google.protobuf.Struct content = 3;
+}
+
+message KeyValueSetResponse {}
+
+message KeyValueDeleteRequest {
+  // Key of the document to delete
+  Key key = 1;
+}
+
+message KeyValueDeleteResponse {}
```

Comment:
Highlighting this for visibility, this was a quick rename change, but any structure we want to change here can be done quickly.